web/modules/custom/weather/src/Plugin/Block/WeatherBlock.php

```diff
@@ -61,13 +69,23 @@ public function getWeatherSetting() {
    * {@inheritdoc}
    */
   public function build() {
-    $city = 'Lutsk';
+    $city = $this->connection->select('weather_info', 't')
+      ->fields('t', ['user_city'])->execute()->fetchAll();
+    foreach ($city as $row) {
+      $city = $row->user_city;
+    }
```

Comment:
In the block and the form, you duplicate the code that receives the city value. It would be better to put this code in the service and call this method where necessary